File: 0435-non-overlapping-intervals/0435-non-overlapping-intervals.js
/**
 * @param {number[][]} intervals
 * @return {number}
 */
var eraseOverlapIntervals = function(intervals) {
    intervals.sort((a, b) => a[0] - b[0])
    var counter = 0;
    var i = 0;
    var j = 1;
    while (i < intervals.length && j < intervals.length) {
        var first = intervals[i];
        var second = intervals[j];
        if (first[1] > second[0]) {
            if (first[1] < second[1]) {
                j++;
            } else {
                j++;
                i = j - 1;
            }
            counter++;
        } else {
            j++;
            i = j - 1;
        }
    }
    return counter;
};

File: 0435-non-overlapping-intervals/0435-non-overlapping-intervals.py
class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        intervals.sort(key=lambda x: x[0])
        counter = 0
        i = 1
        end = intervals[0][1]
        while i < len(intervals):
            if intervals[i][0] < end:
                end = min(end, intervals[i][1])
                counter += 1
            else:
                end = intervals[i][1]
            i += 1
        return counter